Disable ctrl+roller to change editor font size

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Is this feature provided in the master branch?

- [X] I have checked the changelog

### The problem you are facing

I used ctrl+Z/X/... shortcut a lot but sometimes too early to roll code before the ctrl key was released. Then I had to wait a few seconds and open settings to change the font size back.

### Describe the feature you'd like

It solves my problem if checkbox of whether enable ctrl+roller to change editor font size(enabled default) can be added, Thx a lot.

### Describe alternatives you've considered

_No response_

### Are you willing to contribute to this?

- [ ] I'm ready to contribute to this (and will open a PR soon)
- [ ] I'd like to have a try (with the help of the maintainers)
- [X] No, thanks

### Anything else?

_No response_

Hey @ouuan I am facing some issues while building cpeditor from source. I am using `zsh`

``` zsh
cpeditor|fakevim-integration ⇒ mkdir build
cd build
cmake .. -DCMAKE_BUILD_TYPE=Release
cmake --build .
mkdir: cannot create directory ‘build’: File exists
CMake Error at CMakeLists.txt:22 (find_package):
  By not providing "FindQt5.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt5", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt5" with any of
  the following names:

    Qt5Config.cmake
    qt5-config.cmake

  Add the installation prefix of "Qt5" to CMAKE_PREFIX_PATH or set "Qt5_DIR"
  to a directory containing one of the above files.  If "Qt5" provides a
  separate development package or SDK, be sure it has been installed.


-- Configuring incomplete, errors occurred!
gmake: Makefile: No such file or directory
gmake: *** No rule to make target 'Makefile'.  Stop.

build|fakevim-integration ⇒ uname -a
Linux yash 6.5.0-17-generic #17-Ubuntu SMP PREEMPT_DYNAMIC Thu Jan 11 14:01:59 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux


build|fakevim-integration ⇒ neofetch
            .-/+oossssoo+/-.               yash@yash 
        `:+ssssssssssssssssss+:`           --------- 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 23.10 x86_64 
    .ossssssssssssssssssdMMMNysssso.       Host: 82K2 IdeaPad Gaming 3 15ACH6 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 6.5.0-17-generic 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 1 day, 16 hours, 10 mins 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 1703 (dpkg), 142 (brew), 15 (snap) 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: zsh 5.9 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 1920x1080 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   DE: GNOME 45.2 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: Mutter 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   WM Theme: Adwaita 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Theme: Yaru-viridian-dark [GTK2/3] 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Icons: Yaru-viridian [GTK2/3] 
  +sssssssssdmydMMMMMMMMddddyssssssss+     Terminal: kitty 
   /ssssssssssshdmNNNNmyNMMMMhssssss/      CPU: AMD Ryzen 5 5600H with Radeon Graphics (12) @ 4.280GHz 
    .ossssssssssssssssssdMMMNysssso.       GPU: AMD ATI Radeon Vega Series / Radeon Vega Mobile Series 
      -+sssssssssssssssssyyyssss+-         GPU: NVIDIA GeForce GTX 1650 Mobile / Max-Q 
        `:+ssssssssssssssssss+:`           Memory: 4030MiB / 13816MiB 
            .-/+oossssoo+/-.
                                                                   
                                                                   

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,10 @@
 
 -   Now the diff viewer uses the same font as the test cases (monospace by default). (#1396)
 
+### Changed
+
+-   Removed the function to scale the editor font via mouse wheel scroll, as it has been reported to have severe performance issues. (#1249 and #1388)
+
 ## v7.0
 
 In this version, we have switched to [KSyntaxHighlighting](https://api.kde.org/frameworks/syntax-highlighting/html/), the same framework used by [the Kate editor](https://kate-editor.org/). This upgrade results in significantly improved syntax highlighting, a wider selection of [color themes](https://kate-editor.org/themes/), and additional features like code folding. However, due to extensive refactoring, some bugs may arise.

--- a/src/Editor/CodeEditor.cpp
+++ b/src/Editor/CodeEditor.cpp
@@ -440,33 +440,6 @@ bool CodeEditor::vimCursor() const
     return m_vimCursor;
 }
 
-void CodeEditor::wheelEvent(QWheelEvent *e)
-{
-    if (e->modifiers() == Qt::ControlModifier)
-    {
-        const auto sizes = QFontDatabase::standardSizes();
-        if (sizes.isEmpty())
-        {
-            LOG_ERR("QFontDatabase::standardSizes() is empty");
-            return;
-        }
-        int newSize = font().pointSize();
-        if (e->angleDelta().y() > 0)
-            newSize = qMin(newSize + 1, sizes.last());
-        else if (e->angleDelta().y() < 0)
-            newSize = qMax(newSize - 1, sizes.first());
-        if (newSize != font().pointSize())
-        {
-            QFont newFont = font();
-            newFont.setPointSize(newSize);
-            setFont(newFont);
-            emit fontChanged(newFont);
-        }
-    }
-    else
-        QPlainTextEdit::wheelEvent(e);
-}
-
 void CodeEditor::highlightOccurrences()
 {
     occurrencesExtraSelections.clear();

--- a/src/Editor/CodeEditor.hpp
+++ b/src/Editor/CodeEditor.hpp
@@ -136,12 +136,6 @@ class CodeEditor : public QPlainTextEdit
 
     void applySettings(const QString &lang);
 
-  signals:
-    /**
-     * @brief Signal, the font is changed by the wheel event.
-     */
-    void fontChanged(const QFont &newFont);
-
   public slots:
     /**
      * @brief Slot, that indent the selected lines.
@@ -203,11 +197,6 @@ class CodeEditor : public QPlainTextEdit
      * @brief Method, update the bottom margin when the font changes.
      */
     void changeEvent(QEvent *e) override;
-
-    /**
-     * @brief Method, update the font size when the wheel is rotated with Ctrl pressed
-     */
-    void wheelEvent(QWheelEvent *e) override;
 
     /**
      * @brief Method, that's called on any key press, posted

--- a/src/appwindow.cpp
+++ b/src/appwindow.cpp
@@ -426,7 +426,6 @@ void AppWindow::openTab(MainWindow *window, MainWindow *after)
     connect(window, &MainWindow::requestUpdateLanguageServerFilePath, this, &AppWindow::updateLanguageServerFilePath);
     connect(window, &MainWindow::editorLanguageChanged, this, &AppWindow::onEditorLanguageChanged);
     connect(window, &MainWindow::editorTextChanged, this, &AppWindow::onEditorTextChanged);
-    connect(window, &MainWindow::editorFontChanged, this, [this] { onSettingsApplied("Appearance/Font"); });
     connect(window, &MainWindow::requestToastMessage, trayIcon,
             [this](QString const &head, QString const &body) { trayIcon->showMessage(head, body); });
     connect(window, &MainWindow::compileOrRunTriggered, this, &AppWindow::onCompileOrRunTriggered);

--- a/src/mainwindow.cpp
+++ b/src/mainwindow.cpp
@@ -119,7 +119,6 @@ void MainWindow::setEditor()
     ui->editorArea->addWidget(editor);
 
     connect(editor, &Editor::CodeEditor::textChanged, this, &MainWindow::onTextChanged);
-    connect(editor, &Editor::CodeEditor::fontChanged, this, &MainWindow::onEditorFontChanged);
     // cursorPositionChanged() does not imply selectionChanged() if you press Left with
     // a selection (and the cursor is at the begin of the selection)
     connect(editor, &Editor::CodeEditor::cursorPositionChanged, this, &MainWindow::updateCursorInfo);
@@ -1240,12 +1239,6 @@ void MainWindow::onTextChanged()
         autoSaveTimer->start();
     }
     emit editorTextChanged(this);
-}
-
-void MainWindow::onEditorFontChanged(const QFont &newFont)
-{
-    SettingsHelper::setEditorFont(newFont);
-    emit editorFontChanged();
 }
 
 void MainWindow::updateCursorInfo()

--- a/src/mainwindow.hpp
+++ b/src/mainwindow.hpp
@@ -157,7 +157,6 @@ class MainWindow : public QMainWindow
     void onRunKilled(int index);
 
     void onFileWatcherChanged(const QString &);
-    void onEditorFontChanged(const QFont &newFont);
     void onTextChanged();
     void updateCursorInfo();
     void updateChecker();
@@ -178,7 +177,6 @@ class MainWindow : public QMainWindow
     void editorFileChanged();
     void requestUpdateLanguageServerFilePath(MainWindow *window, const QString &path);
     void editorTextChanged(MainWindow *window);
-    void editorFontChanged();
     void confirmTriggered(MainWindow *widget);
     void requestToastMessage(const QString &head, const QString &body);
     void editorLanguageChanged(MainWindow *window);

